src/main/groovy/com/codebrig/arthur/observe/structure/literal/PhpLiteral.groovy

```diff
@@ -9,21 +9,40 @@ import com.codebrig.arthur.observe.structure.StructureLiteral
  * @version 0.4
  * @since 0.2
  * @author <a href="mailto:brandon.fergerson@codebrig.com">Brandon Fergerson</a>
+ * @author <a href="mailto:valpecaoco@gmail.com">Val Pecaoco</a>
  */
 class PhpLiteral extends StructureLiteral {
 
     @Override
     String getNodeLiteralAttribute(SourceNode node) {
-        return null
+        switch (Objects.requireNonNull(node).internalType) {
+            case "Scalar_LNumber":
+                return numberValueLiteral()
+            case "Scalar_DNumber":
+                return doubleValueLiteral()
+            case "Scalar_String":
+                return stringValueLiteral()
+            default:
+                return null
+        }
     }
 
     @Override
     List<String> getPossibleNodeLiteralAttributes(SourceNode node) {
-        def literalAttribute = getNodeLiteralAttribute(node)
-        if (literalAttribute == null) {
-            return Collections.emptyList()
-        } else {
-            return Collections.singletonList(literalAttribute)
+        switch (Objects.requireNonNull(node).internalType) {
+            case "Scalar_LNumber":
+                return [numberValueLiteral()]
+            case "Scalar_DNumber":
+                return [doubleValueLiteral()]
+            case "StringLiteral":
+                return [stringValueLiteral()]
+            default:
+                return Collections.emptyList()
         }
     }
+
+    @Override
+    boolean isNegative(SourceNode node) {
```

Comment:
Ah, here is the other usage of `isNegative()`. Okay, please just rename to `isNodeLiteralNegative()` so it matches the other function names that check the SourceNode.

Reply:
Resolved with commit https://github.com/CodeBrig/Arthur/pull/79/commits/51f0884fa52bfb80b0d83652785a5198d3fff374